### ISSUE TITLE
Define column dependence probability

### ISSUE THREAD

**Issue by axch** [probcomp/bayeslite#268]
- Define it probabilistically
- Implement the definition as an (approximate?) algorithm against joint simulation and assessment
- Compare speed and quality against the structural implementation in crosscat

**Comment by gregory-marton** [comment]
@axch, can you give more context? Isn't dependence probability already defined on columns (aka variables)?

**Comment by riastradh-probcomp** [comment]
I'm not sure we have a formal probability theory definition of it.  One possibility is something like Pr[MI(X, Y) = 0 | H], where H is the metamodel.  In this case, we ought to be able to write a generic Monte Carlo estimator for it using the metamodel's simulate and logpdf methods.

**Comment by axch** [comment]
Yes, we have a standing candidate definition of dependence probability, for arbitrary pairs of cells.  (I interpret "on columns" as "on cells of those columns from one notional fresh row".)  The standing definition in question is

"Probability of dependence of A and B is the probability under the uncertainty in inferring the model that the mutual information of cells A and B is nonzero".

The question of implementing this generically in terms of simulate and assess (presumably approximately) remains open.  The uncertainty over model inference is taken care of by averaging the answers over the available model, but one model is not obvious.  Assessing whether the mutual information of two cells under one particular model is nonzero, using only a finite number of simulation and assessment queries, is impossible exactly, so we have to either approximate or produce a probabilistic answer.  It is not apparent how to do this right.

The obvious candidate is to draw a large number of 1-sample or low-sample estimates of the mutual information and try to decide whether the results of doing that are consistent with the true MI being zero; presumably this can be done up to a p-value with a T-test or Z-score test, but then what?  Do we report 0 or 1 (for that model) if the p-value meets a threshold?  If so, how to choose the threshold?  Otherwise, do we report some probability of non-zeroness?  To do that consistently, we would need a probabilistic model of attempts to compute the mutual information: a prior on possible MI values (Jeffreys?) and a likelihood for getting one estimate of the MI or another.  Of course, if all the quantities in such a model are continuous, the probability of the MI being exactly zero will always be zero, so we need to put in a discrete bias favoring the possibility of independence.  What's the proper setting for that bias?  (Maybe we just call it spike and slab and say the prior probability of the spike is 0.5 because it probably doesn't much matter).
It might also be possible to use a Jaynes-style test: compute the evidence for "0" vs the best single-point alternative and accept if it's not too high; though we still need a model for the results of 1-sample MI estimates (might work to use a Gaussian model for multi-sample ones).
